### PR TITLE
Update README.md to specify supported Python versions

### PR DESCRIPTION
**Pull request by AlexonOliveiraRH**
Updating the documentation to explicitly include the information we only support Python 3.10 and 3.11 so far.

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the
  [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary).
- [ ] [Changelog](https://github.com/instructlab/instructlab/blob/main/CHANGELOG.md) updated with breaking and/or notable changes for the next minor release.
- [x] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Functional tests have been added, if necessary.
- [ ] E2E Workflow tests have been added, if necessary.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -93,7 +93,7 @@ For an overview of the full workflow, see the [workflow diagram](./docs/workflow
 - C++ compiler
 - Python 3.10 or Python 3.11
 
-   📋 Python 3.12 is currently not supported. Some InstructLab dependencies don't work on Python 3.12, yet.
+   📋 Python 3.12 is currently not supported. Some InstructLab dependencies don't work on Python 3.12, yet. As of now, we only support Python 3.10 or Python 3.11. Any other Python version would install an older version of InstructLab (e.g. 0.17).
```
Comment (reidliu41) [inline README.md:96]: Python 3.11 (preferred)

Comment (AlexonOliveiraRH) [reply README.md]: > Python 3.11 (preferred)

@reidliu41 ACK. Done.

**Comment by AlexonOliveiraRH** [comment]
@reidliu41 the following was merged following the same recommendation: https://github.com/instructlab/docs.instructlab.ai/pull/22

Is anything left to do in this one so we can move forward? Thank you in advance.

**Comment by reidliu41** [comment]
@kelbrown20 hi could you please take a look?

**Review by kelbrown20 — APPROVED** [review]
Hello! Updates look good to me, thank you!

**Comment by AlexonOliveiraRH** [comment]
> Hello! Updates look good to me, thank you!

@kelbrown20 @reidliu41 thank you. Anything else left to do?

**Comment by kelbrown20** [comment]
Closing this PR in favor of #2717

The wording here might imply that it may be okay, even though not supported, to use other python versions because it still downloads **a** version of InstructLab (But we don't want users to use 0.17)

Please re-open if there are any further comments, questions, or concerns.

**Pull request closed without merging**